File: 169-majority-element/169-majority-element.cpp
class Solution {
public:
    int majorityElement(vector<int>& nums) {
        
        int x = nums[0];
        int ct = 0;
        
        for(int i=0;i<nums.size();i++){
            if(x==nums[i]){
                ct += 1;
            }else{
                ct--;
                if(ct==0){
                    ct=1;
                    x = nums[i];
                }
            }
        }

        return x;
    }
};

File: 169-majority-element/169-majority-element.py
class Solution:
    def majorityElement(self, nums: List[int]) -> int:
        n = len(nums)
        m = defaultdict(int)
        
        for num in nums:
            m[num] += 1
        
        n = n // 2
        for key, value in m.items():
            if value > n:
                return key
        
        return 0
